API Upgrade file updated after ShippingMethod endpoints changes (#13659)
Sylius/Sylius: UPGRADE-API-1.12.md

# UPGRADE FROM `v1.11.X` TO `v1.12.0`

1. The `Sylius\Bundle\ApiBundle\DataProvider\CartShippingMethodsSubresourceDataProvider` has been changed to `Sylius\Bundle\ApiBundle\DataProvider\ShippingMethodsCollectionDataProvider`.

> 1. The `Sylius/Bundle/ApiBundle/Serializer/ShippingMethodNormalizer` logic and constructor has been changed due to refactor above.

Reviewer: ```suggestion
1. The `Sylius\Bundle\ApiBundle\Serializer\ShippingMethodNormalizer` logic and constructor has been changed due to refactor above.
```

    ```diff
        public function __construct(
            private OrderRepositoryInterface $orderRepository,
            private ShipmentRepositoryInterface $shipmentRepository,
            private ServiceRegistryInterface $shippingCalculators,
    +       private RequestStack $requestStack,
    +       private ChannelContextInterface $channelContext
        ) {
            ...
        }
    ```
   

> 1. The  `GET` `api/v2/shop/orders/{token}/shipments/{id}/methods` and `api/v2/shop/shipments/{id}/methods` endpoints has been removed and changed into collection request with 2 parameters `api/v2/shop/shipping-methods?shipmentId={id}&orderToken={token}`, now when we do not provide parameters in response it returns all available `shippingMethods` in channel. Wrong parameters otherwise causes empty array `[]` in response and correct parameters returns shipping methods available for your `shipment`. Here is how the response look:

Reviewer 2: ```suggestion
1. The  `GET` `api/v2/shop/orders/{token}/shipments/{id}/methods` and `api/v2/shop/shipments/{id}/methods` endpoints have been removed and changed into collection request with 2 parameters `api/v2/shop/shipping-methods?shipmentId={id}&orderToken={token}`, now when we do not provide parameters in response it returns all available `shippingMethods` in channel. Wrong parameters otherwise causes empty array `[]` in response and correct parameters returns shipping methods available for your `shipment`. Here is how the response look:
```

> 1. The  `GET` `api/v2/shop/orders/{token}/shipments/{id}/methods` and `api/v2/shop/shipments/{id}/methods` endpoints have been removed and changed into collection request with 2 parameters `api/v2/shop/shipping-methods?shipmentId={id}&orderToken={token}`, now when we do not provide parameters in response it returns all available `shippingMethods` in channel. Wrong parameters otherwise causes empty array `[]` in response and correct parameters returns shipping methods available for your `shipment`. Here is how the response look:

Reviewer: ```suggestion
1. The  `GET` `api/v2/shop/orders/{token}/shipments/{id}/methods` and `api/v2/shop/shipments/{id}/methods` endpoints have been removed and changed into collection request with 2 parameters `api/v2/shop/shipping-methods?shipmentId={id}&orderToken={token}`, now when we do not provide parameters in response it returns all available `shippingMethods` in channel. Wrong parameters otherwise cause empty array `[]` in response and correct parameters return shipping methods available for your `shipment`. Here is what the response looks like:
```
And divide this line into several lines

# UPGRADE FROM `v1.11.X` TO `v1.12.0`

> 1. The `Sylius\Bundle\ApiBundle\DataProvider\CartShippingMethodsSubresourceDataProvider` has been changed to `Sylius\Bundle\ApiBundle\DataProvider\ShippingMethodsCollectionDataProvider`.

Reviewer 3: ```suggestion
1. The `Sylius\Bundle\ApiBundle\DataProvider\CartShippingMethodsSubresourceDataProvider` has been removed and replaced by `Sylius\Bundle\ApiBundle\DataProvider\ShippingMethodsCollectionDataProvider`.
```
WDYT?

Here is how the response looks like:
   
   ```
      {
        "@context": "/api/v2/contexts/ShippingMethod",
        "@id": "/api/v2/shop/shipping-methods",
        "@type": "hydra:Collection",
        "hydra:member": [
          {
            "@id": "/api/v2/shop/shipping-methods/ups",
            "@type": "ShippingMethod",
            "id": 1,
            "code": "ups",
            "position": 0,
            "name": "UPS",
            "description": "Quasi perferendis debitis officiis ut inventore exercitationem."
          }
        ],
        "hydra:totalItems": 1,
        "hydra:search": {
          "@type": "hydra:IriTemplate",
          "hydra:template": "/api/v2/shop/shipping-methods{?shipmentId,tokenValue}",
          "hydra:variableRepresentation": "BasicRepresentation",
          "hydra:mapping": [
            {
              "@type": "IriTemplateMapping",
              "variable": "shipmentId",
              "property": null,
              "required": false
            },
            {
              "@type": "IriTemplateMapping",
              "variable": "tokenValue",
              "property": null,
              "required": false
            }
          ]
        }

>       }

Reviewer 3: IMHO this example is not necessary
